Squest/utils/squest_api_views.py

```diff
@@ -0,0 +1,140 @@
+from rest_framework.exceptions import APIException, PermissionDenied, NotAuthenticated
+from rest_framework.generics import GenericAPIView, ListCreateAPIView, RetrieveUpdateDestroyAPIView, \
+    RetrieveUpdateAPIView, ListAPIView, CreateAPIView, RetrieveAPIView, DestroyAPIView
+from rest_framework.permissions import IsAuthenticated, DjangoObjectPermissions, SAFE_METHODS
+from rest_framework.renderers import BrowsableAPIRenderer
+
+
+class SquestObjectPermissions(DjangoObjectPermissions):
+
+    def has_permission(self, request, view):
+        # Skip Class based permissions, otherwise we need to assign user permissions to all ours users.
+        return True
+
+    def has_object_permission(self, request, view, obj):
+        # authentication checks have already executed via has_permission
+        queryset = self._queryset(view)
+        model_cls = queryset.model
+        user = request.user
+
+        perms = self.get_required_object_permissions(request.method, model_cls)
```

Comment:
It could be better to override get_required_object_permissions as it:
``` python
    def get_required_object_permissions(self, method, model_cls):
        kwargs = {
            'app_label': model_cls._meta.app_label,
            'model_name': model_cls._meta.model_name
        }

        if method not in self.perms_map:
            raise exceptions.MethodNotAllowed(method)

        return [perm % kwargs for perm in self.get_perms_map()[method]]
```
with get_perms_map by default:
``` python
def get_perms_map(self):
    return self.perms_map
```
With both it is possible to override perms_map easily each API view.

Reply:
Not feasible this way, keep in mind